wasm/pow/argon2id/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+use anubis::{DATA_BUFFER, DATA_LENGTH, update_nonce};
+use argon2::Argon2;
+use std::boxed::Box;
+use std::sync::{LazyLock, Mutex};
+
+/// SHA-256 hashes are 32 bytes (256 bits). These are stored in static buffers due to the
+/// fact that you cannot easily pass data from host space to WebAssembly space.
+pub static RESULT_HASH: LazyLock<Mutex<[u8; 32]>> = LazyLock::new(|| Mutex::new([0; 32]));
+
+pub static VERIFICATION_HASH: LazyLock<Box<Mutex<[u8; 32]>>> =
+    LazyLock::new(|| Box::new(Mutex::new([0; 32])));
+
+/// Core validation function. Compare each bit in the hash by progressively masking bits until
+/// some are found to not be matching.
+///
+/// There are probably more clever ways to do this, likely involving lookup tables or something
```

Comment:
You should use the Count Leading Zeros intrinsic.

Reply:
- As highlighted in #106, hash validation is simply testing that the given hash is smaller than a given reference depending on difficulty (e.g. difficulty 4 is `hash < 0x00010...` or `hash <= 0x0000f...`). See also:
  - https://github.com/TecharoHQ/anubis/issues/106#issuecomment-2822493966
  - https://github.com/TecharoHQ/anubis/issues/106#issuecomment-2832473776